.travis.yml

```diff
@@ -292,28 +166,39 @@ before_install:
       export CXX="${CXX}-${VERSION}"
     fi
   - export CXXFLAGS="${CXXFLAGS} -Wall -Wextra -Werror -pedantic-errors"
+  - CXX_VERSIONS=(${CXX_VERSIONS//;/ })
+  - BUILD_TYPES=(Debug Release)
 
 install:
-  - mkdir build-debug
-  - cd build-debug
-  - cmake -DCMAKE_BUILD_TYPE=Debug ..
-  - cd ..
-
-  - mkdir build-release
-  - cd build-release
-  - cmake -DCMAKE_BUILD_TYPE=Release ..
-  - cd ..
+  - |
+    for CXX_VERSION in "${CXX_VERSIONS[@]}"; do
+      for BUILD_TYPE in "${BUILD_TYPES[@]}"; do
+        BUILD_DIR="build-${CXX_VERSION}-${BUILD_TYPE}"
+        mkdir "${BUILD_DIR}"
+        pushd "${BUILD_DIR}"
+        (
+          set -ex
+          CXXFLAGS="${CXXFLAGS} -std=c++${CXX_VERSION}" cmake \
+              -DCMAKE_BUILD_TYPE=${BUILD_TYPE} ..
```

Comment:
Instead of splitting the line, just export `CXXFLAGS` on its own line?

Reply:
Yep! I was worried about spilling over state to the next loop, but I think it's protected by the parens here.